lib/modules/platform/github/massage-markdown-links.ts

```diff
@@ -11,7 +11,7 @@ interface UrlMatch {
 }
 
 const urlRegex =
-  /(?:https?:)?(?:\/\/)?(?:www\.)?(?<!api\.)(?:to)?github\.com\/[-_a-z0-9]+\/[-_a-z0-9.]+\/(?:discussions|issues|pull)\/[0-9]+(?:#[-_a-z0-9]+)?/i; // TODO #12872 (?<!re) after text not matching
+  /(?:https?:)?(?:\/\/)?(?:www\.)?(?<!api\.)(?:to)?github\.com\/[-\w\d]+\/[-_.\w\d]+\/(?:discussions|issues|pull)\/\d+(?:#[-_\w\d]+)?/i; // TODO #12872 (?<!re) after text not matching
```

Comment:
Isn't `\w` shortcut for `[A-Za-z0-9_]`?

```suggestion
  /(?:https?:)?(?:\/\/)?(?:www\.)?(?<!api\.)(?:to)?github\.com\/[-\w]+\/[-.\w]+\/(?:discussions|issues|pull)\/\d+(?:#[-\w]+)?/i; // TODO #12872 (?<!re) after text not matching
```

Reply:
I actually prefer the current readability though

Reply:
its not accurate then, 
well how about we remove the useless `(?<!api\.)` ?